spec/rspec/core/formatters_spec.rb

```diff
@@ -130,28 +130,47 @@ module RSpec::Core::Formatters
       end
     end
 
-    describe "#setup_default", "with profiling enabled" do
+    describe "#setup_default" do
       let(:setup_default) { loader.setup_default output, output }
 
-      before do
-        allow(RSpec.configuration).to receive(:profile_examples?) { true }
+      context "with a formatter that implements #message" do
+        it 'doesnt add a fallback formatter' do
+          allow(reporter).to receive(:registered_listeners).with(:message) { [:json] }
+          setup_default
+          expect(loader.formatters).to exclude(an_instance_of ::RSpec::Core::Formatters::MessageFormatter)
+        end
       end
 
-      context "without an existing profile formatter" do
-        it "will add the profile formatter" do
-          allow(reporter).to receive(:registered_listeners).with(:dump_profile) { [] }
+      context "without a formatter that implements #message" do
+        it 'adds a fallback for message output' do
+          allow(reporter).to receive(:registered_listeners).with(:message) { [] }
           setup_default
-          expect(loader.formatters.last).to be_a ::RSpec::Core::Formatters::ProfileFormatter
+          expect(loader.formatters.last).to be_a ::RSpec::Core::Formatters::MessageFormatter
         end
       end
 
-      context "when a formatter that implement #dump_profile is added" do
-        it "wont add the profile formatter" do
-          allow(reporter).to receive(:registered_listeners).with(:dump_profile) { [:json] }
-          setup_default
-          expect(
-            loader.formatters.map(&:class)
-          ).to_not include ::RSpec::Core::Formatters::ProfileFormatter
+      context "with profiling enabled" do
+        before do
+          allow(reporter).to receive(:registered_listeners).with(:message) { [:json] }
```

Comment:
Seems brittle to stub `reporter.registered_listeners` when it's not a public API.  Why not use `reporter.register_listener(spy, :message)`?  That uses a public API (which is going to be more stable as it's subject to SemVer!), doesn't fake things out unnecessarily, and won't have problems with the fact that `:json` isn't actually a listener since `spy` will gladly no-op listen to any message :).

Or do you see an advantage to stubbing the reporter here over just registering a listener?